migrations/20231210123554_init.sql

```diff
@@ -0,0 +1,314 @@
+    -------------------------------------------------------
+    CONSTRAINT check_event_stuff_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE task
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    title       VARCHAR(255) NOT NULL,
+    description TEXT,
+    finished_at DATE,
+    priority    task_priority NOT NULL,
+    accepted    acceptance_status NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    event_id    UUID,
+    creator_id  UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (event_id) REFERENCES event (id),
+    FOREIGN KEY (creator_id) REFERENCES event_staff (id),
+    -------------------------------------------------------
+    CONSTRAINT check_event_title_len
+        CHECK (char_length(title) >= 1),
+    CONSTRAINT check_event_stuff_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE assigned_staff
+(
+    status      acceptance_status NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    task_id     UUID,
+    user_id     UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (task_id) REFERENCES task (id),
+    -------------------------------------------------------
+    CONSTRAINT check_assigned_staff_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE comment
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    content     TEXT NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    event_id    UUID,
+    author_id   UUID,
+    task_id     UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (event_id) REFERENCES event (id),
+    FOREIGN KEY (author_id) REFERENCES "user" (id),
+    FOREIGN KEY (task_id) REFERENCES task (id),
+    -------------------------------------------------------
+    CONSTRAINT check_comment_single_relation_only
+        CHECK (
+            (CASE WHEN event_id IS NULL THEN 0 ELSE 1 END
+             + CASE WHEN task_id IS NULL THEN 0 ELSE 1 END
+            ) = 1),
+    CONSTRAINT check_comment_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+
+);
+
+
+CREATE TABLE associated_company
```

Comment:
Either I'm blind, or there is no PK here. 
Make company_id and event_id the primary key, please.